aidial_adapter_vertexai/llm/bison_adapter.py

```diff
@@ -1,71 +1,170 @@
     BisonChatCompletionAdapter,
 )
 from aidial_adapter_vertexai.llm.bison_prompt import BisonPrompt
 from aidial_adapter_vertexai.llm.exceptions import ValidationError
+from aidial_adapter_vertexai.llm.vertex_ai import (
+    get_chat_model,
+    get_code_chat_model,
+    init_vertex_ai,
+)
 from aidial_adapter_vertexai.universal_api.request import ModelParameters
 
 
-class BisonChatAdapter(BisonChatCompletionAdapter):
-    @override
-    def _create_instance(self, prompt: BisonPrompt) -> Dict[str, Any]:
-        return {
-            "context": prompt.context or "",
-            "messages": prompt.messages,
-        }
+class CodeChatParamsBase(TypedDict, total=False):
+    max_output_tokens: Optional[int]
+    temperature: Optional[float]
 
-    @override
-    def _create_parameters(self, params: ModelParameters) -> Dict[str, Any]:
-        # See chat playground: https://console.cloud.google.com/vertex-ai/generative/language/create/chat
-        ret: Dict[str, Any] = {}
 
-        if params.max_tokens is not None:
-            ret["maxOutputTokens"] = params.max_tokens
+class ChatParamsBase(TypedDict, total=False):
+    max_output_tokens: Optional[int]
+    temperature: Optional[float]
 
-        if params.temperature is not None:
-            ret["temperature"] = params.temperature
+    # Extra compared to CodeChatParams
+    stop_sequences: Optional[List[str]]
+    top_k: Optional[int]
+    top_p: Optional[float]
 
-        if params.stop is not None:
-            ret["stopSequences"] = (
-                [params.stop] if isinstance(params.stop, str) else params.stop
-            )
 
-        if params.top_p is not None:
-            ret["topP"] = params.top_p
+ChatParamsStream = ChatParamsBase
+CodeChatParamsStream = CodeChatParamsBase
 
-        return ret
 
+class NoStreamParams(TypedDict, total=False):
+    candidate_count: Optional[int]
 
-class BisonCodeChatAdapter(BisonChatCompletionAdapter):
-    @override
-    def _create_instance(self, prompt: BisonPrompt) -> Dict[str, Any]:
-        if prompt.context is not None:
-            raise ValidationError("System message is not supported")
 
+class ChatParamsNoStream(ChatParamsBase, NoStreamParams):
+    pass
+
+
+class CodeChatParamsNoStream(CodeChatParamsBase, NoStreamParams):
+    pass
+
+
+class BisonChatAdapter(BisonChatCompletionAdapter):
+    model: ChatModel
+
+    @classmethod
+    async def create(
+        cls, model_id: str, project_id: str, location: str
+    ) -> "BisonChatAdapter":
+        await init_vertex_ai(project_id, location)
+        return cls(await get_chat_model(model_id))
+
+    def prepare_parameters_no_stream(
+        self, params: ModelParameters
+    ) -> ChatParamsNoStream:
         return {
-            "messages": prompt.messages,
+            "max_output_tokens": params.max_tokens,
+            "temperature": params.temperature,
+            "stop_sequences": params.stop,
+            "top_p": params.top_p,
+            "candidate_count": params.n,
```

Comment:
The method send_message_async always returns the first candidate only:
```python
yield response.text
```

Reply:
Indeed, fixed.